salt/modules/win_groupadd.py

```diff
@@ -374,3 +374,32 @@ def __fixlocaluser(username):
         username = ('{0}\\{1}').format(__salt__['grains.get']('host'), username)
 
     return username.lower()
+
+
+def list_groups(refresh=False):
+    '''
+    Return a list of groups
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' group.getent
+    '''
+    if 'group.list_groups' in __context__ and not refresh:
+        return __context__['group.getent']
```

Comment:
It's something Erik added to put these values in `__context__` instead of running the command every time. If it finds it in `__context__`, it will return that list, unless you set `refresh=True`. If it does run the actual command it is added to or updates the value in `__context__` just before it returns the value... see below.